### PR TITLE
Refactoring adding init

### PR DESCRIPTION
**Pull request by PGLongo**
Maybe this PR is not ready to be merge, but I've done a general refactoring moving 

* imageView
* titleLabel
* subtitleLabel
* layoutSubviews

to **PKHUDSquareBaseView** and with a new init with optional values is possible to create all the possible combination of HUD (for example PKHUDSuccessView with title, only subtitle of both).
simulator screen shot 18 gen 2016 21 44 29
simulator screen shot 18 gen 2016 21 46 31
simulator screen shot 18 gen 2016 21 47 04
simulator screen shot 18 gen 2016 21 47 25
simulator screen shot 18 gen 2016 21 49 14

Some classes (PKHUDTitleView, PKHUDSubtitleView, PKHUDStatusView, PKHUDStatusProgressView) became obsolete but I think that only **PKHUDStatusProgressView** should be removed for compatibility reason.

### CODE REVIEW

File: Demo/DemoViewController.swift [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//
+    //
```
Comment (pkluz) [inline Demo/DemoViewController.swift]: Looks like an accidental indent right here :)

**Comment by pkluz** [comment]
Very cool work. This was on my plate for far too long and now you've done it. Very cool. Before I merge this, could you do me one favor though and create a `CONTRIBUTORS` file in the root of the directory, and add the name + email (if available) of all who have contributed to this repo?

**Comment by PGLongo** [comment]
I've declared as a WIP because some files can be deleted and I want discuss it with you.

Do you agree with me to remove only **PKHUDStatusProgressView**?

**Comment by pkluz** [comment]
So I actually like being fairly aggressive about getting things forward even if that means breaking something along the way. It's only gonna hurt more down the road than it would if we did it right now.

Obviously we'll need a version bump to 3.0 since we're breaking API contracts.

But keeping too much stuff around in such a simple component only increases surface area unnecessarily and makes things complex than they need to be for new users.

So since we'll already be breaking compatibility by removing PKHUDStatusProgressView, we might as well go ahead an remove all the others that are redundant in favor of the convenience initializers.

**Comment by PGLongo** [comment]
PKHUDStatusProgressView has been introduced recently and never been added to Cocapods for that reason I suggest the removal, it's not necessary. 

If you want to bump a new version breaking the compatibility we can also add the Determinate Progress Bar as mentioned here (#51, by the way I'm still waiting for an answer) and develop other improvements.

**Comment by PGLongo** [comment]
I've removed the not necessary class.  Do you think is necessary the file CONTRIBUTORS? All the information are available at this link [CONTRIBUTORS](https://github.com/pkluz/PKHUD/graphs/contributors)

**Comment by PGLongo** [comment]
@pkluz please can you answer to my questions? There are a lot of things to do and there are other PR like #69 (it seems a very good PR) that could case conflicts with this.

Thanks

**Comment by Flar49** [comment]
@PGLongo 
It will definitely cause conflicts. I'll adapt the wrapper once this PR is merged.

**Comment by PGLongo** [comment]
Ok @Flar49. By the way great job with your PR!

**Comment by Flar49** [comment]
Thanks ;)

**Comment by pkluz** [comment]
@Flar49 All right let's stick to the GitHub contributors only. Regarding the spinner I'm not sure I want to pull in a dependency - it is pretty however... Let me get back to you on that - will merge this PR over the weekend though.

**Comment by PGLongo** [comment]
Ok, great for the merge.

Discussing about dependency, I've some doubts because I don't like to reinvent the wheel and for that reason I'm using a lot of Pods but that kind of view should be easy to implement so we can also avoid the dependency.

I will make some tries in the weekend.

**Comment by PGLongo** [comment]
@pkluz  congratulations for the 100th commit and the great merge work you made today!

**Comment by pkluz** [comment]
@PGLongo Thanks, and sorry again for the immense delay... I blame work ;-)

**Comment by PGLongo** [comment]
Don't worry. I can understand

**Comment by pkluz** [comment]
Working on getting #69 updated + merged. Gonna bump spec after. Prolly bump to 3.0 as I've been breaking things along the way. And the introduction of a secondary - simpler - api might require it.

**Comment by PGLongo** [comment]
Sure, also for me bumping 3.0 is the best solution.

**Pull request merged**